sycl/test-e2e/Graph/Explicit/buffer_copy_target2host_offset.cpp

```diff
@@ -0,0 +1,55 @@
+// REQUIRES: level_zero, gpu
+// RUN: %{build} -o %t.out
+// RUN: %{run} %t.out
+// RUN: %if ext_oneapi_level_zero %{env ZE_DEBUG=4 %{run} %t.out 2>&1 | FileCheck %s %}
+//
+// CHECK-NOT: LEAK
+
+// Tests adding buffer copy with offset
+// -- Target to Host (read path) --  nodes
+// using the explicit API and submitting
+// the graph.
+// The second run is to check that there are no leaks reported with the embedded
+// ZE_DEBUG=4 testing capability.
+
+#include "../graph_common.hpp"
+
+int main() {
+  queue Queue;
+
+  using T = int;
+
+  int Offset = 100;
+  T DataA[Size], DataB[Size];
+  std::iota(DataA, DataA + Size, 1);
+  std::iota(DataB, DataB + Size, 1000);
+
+  T ReferenceA[Size], ReferenceB[Size];
+  for (size_t i = 0; i < Size; i++) {
+    ReferenceA[i] = DataA[i];
+    if (i < (Size - Offset))
+      ReferenceB[i] = DataA[i + Offset];
+    else
+      ReferenceB[i] = DataB[i];
+  }
+
+  {
+    exp_ext::command_graph Graph{Queue.get_context(), Queue.get_device()};
+
+    buffer<T, 1> BufferA(DataA, range<1>(Size));
+
```

Comment:
I think this test should be consistent with the others in that we should disable copy back on the buffer (and can remove the scope it's all contained in, since that was used to trigger copy back historically) and use a `host_accessor` to get the data back.

Also applies to a few other tests that have this issue.